### ISSUE TITLE
a way to open spotify links

### ISSUE THREAD

**Issue by treeshateorcs** [aome510/spotify-player#218]
**Is your feature request related to a problem? Please describe.**
there's currently no way to open spotify links.

**Describe the solution you'd like**
search page should recognize spotify urls and act accordingly

**Describe alternatives you've considered**
 i tried putting a link into search but it didn't give the result i wanted (and i also couldn't find the playlist by name)

**Additional context**
example of url - https://open.spotify.com/playlist/0Hm7mXPldSTuokTO7eqtVU?si=VU6LBFIVQkCMePwM26_rPA

**Comment by aome510** [comment]
Let's me think about how this feature should be implemented.

> (and i also couldn't find the playlist by name)

I can search your example playlist just fine.

<img width="964" alt="Screenshot 2023-07-15 at 3 21 46 PM" src="https://github.com/aome510/spotify-player/assets/40011582/dec4bbf2-937d-4196-b641-ac98abf6bb88">

**Comment by jacksongoode** [comment]
Its interesting because Spotify will detect urls in their search. Maybe you can do some regex to determine if its a url and then return the relevant results.

**Comment by aome510** [comment]
> Its interesting because Spotify will detect urls in their search. Maybe you can do some regex to determine if its a url and then return the relevant results (whether the url contains "/tracks/" use the track API, etc).

yeah, that's what I have in mind to implement this feature. Unfortunately, I'm kinda busy nowadays, so I'm currently doing very basic maintaining work for this project.
